components/browser/state/src/main/java/mozilla/components/browser/state/action/BrowserAction.kt

```diff
@@ -614,3 +616,28 @@ sealed class ContainerAction : BrowserAction() {
      */
     data class RemoveContainerAction(val contextId: String) : ContainerAction()
 }
+
+/**
+ * [BrowserAction] implementations related to updating search engines in [SearchState].
+ */
+sealed class SearchAction : BrowserAction() {
+    /**
+     * Updates [BrowserState.search] to add/modify [SearchState.searchEngines].
+     */
+    data class UpdateSearchEngineListAction(val searchEngineList: List<SearchEngine>) : SearchAction()
```

Comment:
Naming nit regarding `Update` (also for action below): Maybe we should make it explicit and name it `Add` or `Set` and only do that. Actions that potentially do two things can cause issues later.